src/tox/tox_env/python/pip/req/file.py

```diff
@@ -57,9 +57,15 @@ def __init__(self, req: str, options: dict[str, Any], from_file: str, lineno: in
                 else:
                     path = root / req
                 extra_part = f"[{','.join(sorted(extras))}]" if extras else ""
-                rel_path = str(path.resolve().relative_to(root))
-                if rel_path != "." and os.sep not in rel_path:  # prefix paths in cwd to not convert them to requirement
-                    rel_path = f".{os.sep}{rel_path}"
+                try:
+                    rel_path = str(path.resolve().relative_to(root))
+                    if (
+                        rel_path != "." and os.sep not in rel_path
+                    ):  # prefix paths in cwd to not convert them to requirement
+                        rel_path = f".{os.sep}{rel_path}"
+                except ValueError:
```

Comment:
Can we guard this different behavior with a version filter so we remember to remove it once the CPython fix lands in our lowest supported version?

Reply:
They do not have the plan to fix the issue in stdlib, only the error message would be different. Basically they will document that `relative_to` is never expected to produce paths that start with `../`, instead it would fail.

Reply:
Fair, tests still in place.